### PR TITLE
Award player an achievement when creating a beacon

### PR DESCRIPTION
**Pull request by Bond-009**
https://minecraft.gamepedia.com/Beacon#Achievements

### CODE REVIEW

**Comment by bearbin** [comment]
Clang generates an error in the build - see the travis failure.

**Comment by lkolbly** [comment]
I haven't tried it to be sure, but shouldn't it only award the achievement if m_BeaconLevel == 4? (or whatever the highest is)

https://minecraft.gamepedia.com/Achievements

**Comment by williamhatcher** [comment]
> Be within a 20×20×14 cuboid centered on a beacon block when it realizes it has become powered

I think you're right!

**Comment by lkolbly** [comment]
Huh, interesting, so there's just the one achievement, "The Beaconator", which is for creating a full pyramid, but the advancements split it into two - "Bring Home the Beacon" which only requires turning on a beacon and "Beaconator" which is the full beacon.

**Comment by Bond-009** [comment]
`The player can achieve the Beaconator achievement by placing a beacon on a level-4 pyramid of specific mineral blocks.`
You are right.
Ref: https://minecraft.gamepedia.com/Beacon#History

File: src/BlockEntities/BeaconEntity.cpp [hunk 1]
```diff
@@ -187,16 +187,42 @@ void cBeaconEntity::UpdateBeacon(void)
...
+				{
+					a_Player->AwardAchievement(eStatistic::achFullBeacon);
+				}
+				return false;
+			}
+
+		private:
+			Vector3d m_Position;
+		} PlayerCallback(Vector3d(m_PosX, m_PosY, m_PosZ));
+		GetWorld()->ForEachPlayer(PlayerCallback);
```
Comment (madmaxoft) [inline src/BlockEntities/BeaconEntity.cpp]: We really need to invest the time to turn these into lambdas...

**Pull request merged**